docs/product-user-guide/src/main/asciidoc/drl-rules-WHEN-proc.adoc

```diff
@@ -0,0 +1,160 @@
+[id='drl-rules-WHEN-proc']
+= Adding WHEN conditions in DRL rules
+
+The `when` part of the rule contains the conditions that must be met to execute an action. For example, if a bank requires loan applicants to have over 21 years of age, then the `when` condition of an "Underage" rule would be `Applicant( age < 21 )`. Conditions consist of a series of stated patterns and constraints, with optional bindings and other supported Java elements, based on the available data objects in the package.
+
+.Prerequisites
+* The `package` is defined at the top of the DRL file. This should have been done for you when you created the file.
+* The `import` list of data objects used in the rule is defined below the `package` line of the DRL file. Data objects can be from this package or from another package in {CENTRAL}.
+* The `rule` name is defined in the format `rule "name"` below the `package`, `import`, and other lines that apply to the entire DRL file. The same rule name cannot be used more than once in the same package. Optional rule attributes (such as `salience` or `no-loop`) that define rule behavior are below the rule name, before the `when` section.
+
+.Procedure
+. In the DRL designer, enter `when` within the rule to begin adding condition statements. The `when` section consists of zero or more fact patterns that define conditions for the rule.
++
+If the `when` section is empty, then actions in the `then` section are executed once every time the decision engine creates a new session. This is useful when certain actions apply to all rules every session, regardless of conditions. *<@Michael (7 of 13): I tried to make this clearer, but not sure I'm right. This close? And again, is "decision engine" interchangeable with "working memory" of the decision engine, just for the sake of simplicity?>*
```

Comment:
@manstis , what you say here and what I think was in the Drools doc is what I'm trying to convey: "Since the rule has no LHS it will always generate an activation and execute." The "always" is what I'm trying to be more specific about. Every time what happens? Every time fireAllRules() is invoked you say, but when does that occur? And in what cases is this type of rule helpful? Basically, I just want to tell the use what happens when the LHS is empty (and saying it "always" fires is not specific enough, to me) and when they should consider using this type of rule (why it's useful).

Reply:
@sterobin The rule, without a LHS, will activate when ```fireAllRules()``` is called; **not** when a session is created (i.e. with ```KieSession kieSession = kiebase.newKieSession()```).

Reply:
@sterobin I give some examples where I talk about "bootstrap".. it's useful when the User wants to "set up" the Decision Engine state using rules (as opposed to using Java code). There may be other common uses of which I am unaware... @mariofusco any other common uses of empty LHSs?

Reply:
@manstis , this work? "If the `when` section is empty, then actions in the `then` section are executed every time a `fireAllRules()` call is made in the decision engine. This is useful if you want to use rules to set up the decision engine state."

I know "in the decision engine" is probably vague, since it's technically the KIE session, I think, but I'm trying to keep it general. If anyone else responds here, I can update. Just want something that's at least accurate, for now.

Reply:
@sterobin Yes, it works for me.. I have one nagging doubt about whether it is 100% accurate for both "stateful" and "stateless" sessions. IIRC rules in a "stateful" session do not re-activate on subsequent ```fireAllRules()``` unless there is new/changed data in the Decision Engine to justify further activation. IDK whether the same applies for a "stateful" session and empty LHSs.. i.e. IDK if the rules would re-activate for subsequent calls to ```fireAllRules()```. Another for the engine expert @mariofusco to comment on.